tpu/src/main/java/tpu/CreateSpotQueuedResource.java

```diff
@@ -0,0 +1,104 @@
+ * limitations under the License.
+ */
+
+package tpu;
+
+// [START tpu_queued_resources_create_spot]
+import com.google.cloud.tpu.v2alpha1.CreateQueuedResourceRequest;
+import com.google.cloud.tpu.v2alpha1.Node;
+import com.google.cloud.tpu.v2alpha1.QueuedResource;
+import com.google.cloud.tpu.v2alpha1.SchedulingConfig;
+import com.google.cloud.tpu.v2alpha1.TpuClient;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public class CreateSpotQueuedResource {
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Google Cloud project you want to create a node.
+    String projectId = "YOUR_PROJECT_ID";
+    // The zone in which to create the TPU.
+    // For more information about supported TPU types for specific zones,
+    // see https://cloud.google.com/tpu/docs/regions-zones
+    String zone = "us-central1-f";
+    // The name for your TPU.
+    String nodeName = "YOUR_TPU_NAME";
+    // The accelerator type that specifies the version and size of the Cloud TPU you want to create.
+    // For more information about supported accelerator types for each TPU version,
+    // see https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#versions.
+    String tpuType = "v2-8";
+    // Software version that specifies the version of the TPU runtime to install.
+    // For more information see https://cloud.google.com/tpu/docs/runtimes
+    String tpuSoftwareVersion = "tpu-vm-tf-2.14.1";
+    // The name for your Queued Resource.
+    String queuedResourceId = "QUEUED_RESOURCE_ID";
+
+    createQueuedResource(
+        projectId, zone, queuedResourceId, nodeName, tpuType, tpuSoftwareVersion);
+  }
+
+  // Creates a Queued Resource with --preemptible flag.
+  public static QueuedResource createQueuedResource(
+      String projectId, String zone, String queuedResourceId,
+      String nodeName, String tpuType, String tpuSoftwareVersion)
+      throws IOException, ExecutionException, InterruptedException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (TpuClient tpuClient = TpuClient.create()) {
+      String parent = String.format("projects/%s/locations/%s", projectId, zone);
+      SchedulingConfig schedulingConfig = SchedulingConfig.newBuilder()
+          .setPreemptible(true)
+          .build();
+
+      Node node =
+          Node.newBuilder()
+              .setName(nodeName)
+              .setAcceleratorType(tpuType)
+              .setRuntimeVersion(tpuSoftwareVersion)
+              .setSchedulingConfig(schedulingConfig)
+              .setQueuedResource(
+                  String.format(
+                      "projects/%s/locations/%s/queuedResources/%s",
+                      projectId, zone, queuedResourceId))
```

Comment:
We want to keep the code consistent. Consider to replace with

```suggestion
      var parent = String.format("projects/%s/locations/%s", projectId, zone);
      var resourceName = String.format("projects/%s/locations/%s/queuedResources/%s",
                      projectId, zone, queuedResourceId);
      SchedulingConfig schedulingConfig = SchedulingConfig.newBuilder()
          .setPreemptible(true)
          .build();

      Node node =
          Node.newBuilder()
              .setName(nodeName)
              .setAcceleratorType(tpuType)
              .setRuntimeVersion(tpuSoftwareVersion)
              .setSchedulingConfig(schedulingConfig)
              .setQueuedResource(resourceName)
```

Reply:
Thank you! Fixed